### PR TITLE
fix(backend): PKPM SATWE modeling fixes for steel frame analysis

### LINKED ISSUE #50: feat(skills): define analysis engine skills for OpenSees, PKPM, and YJK
### Body / 正文

This issue defines analysis engine skills that adapt the unified structural format into engine-specific inputs and trigger execution. OpenSees should be covered first, with clear extension points for PKPM and YJK.

这个 issue 用于定义计算引擎技能，将统一结构格式转换为引擎专用输入并触发执行。应优先覆盖 OpenSees，并为 PKPM 和 YJK 预留清晰扩展点。

### Checklist / 清单

- Define common engine skill contracts
- Split common adapter behavior from engine-specific logic
- Define execution lifecycle, status reporting, and result collection

- 定义统一的引擎技能协议
- 拆分公共适配逻辑与引擎特有逻辑
- 定义执行生命周期、状态反馈与结果收集方式

### Done When / 完成标准

- Engine skills share a stable contract
- OpenSees is covered first, with clear extension points for PKPM and YJK
- Result collection and failure handling are documented

- 引擎技能共享稳定协议
- 优先覆盖 OpenSees，并为 PKPM、YJK 预留清晰扩展点
- 结果收集与失败处理方式已文档化

### PR DESCRIPTION
**Pull request by guyi2000**
## Summary

- **Problem:** PKPM SATWE analysis for steel frame models had 5 bugs causing crashes, wrong I-section field mapping, and incorrect material labels. Custom H-sections couldn't be parsed, and the material type showed "钢砼结构" instead of "钢结构".
- **Why it matters:** Steel frame analysis was either crashing or producing incorrect results, blocking real engineering workflows.
- **What changed:** Fixed AddStandFloor crash, corrected I-section field mapping (B=tw, U=bf, T=tf, D=bf, F=tf), added custom H-section parsing, set section material via `Set_M(5)`, implemented JWSCYCLE execution via DirectorySet.conf, and added post-processing to correct the material label in output files.
- **What did NOT change (scope boundary):** Frontend, other analysis engines (OpenSees), chat/session logic, agent orchestration — only `pkpm-static/` converter, runtime, and `frame/` model builder were modified.

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [ ] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [ ] Gateway / orchestration
- [x] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [x] API / contracts
- [ ] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #
- Related #
- [x] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

- **Root cause:** PKPM I-section field mapping was incorrect (B was being set to flange width instead of web thickness). AddStandFloor() was being called when it causes crashes with beams. JWSCYCLE auto-detects material by scanning sections and classifies custom sections (SetUserSect) as "钢砼" regardless of Set_M(5) and KIND 103=10303.
- **Missing detection / guardrail:** No validation that PKPM API field mapping matched the official APIPythonTest.py reference.
- **Prior context:** I-section mapping was originally guessed; official example was found later in API docs.
- **Why this regressed now:** Initial PKPM converter was written without access to official API documentation.

## Regression Test Plan (if applicable)

- Coverage level that should have caught this:
  - [x] Unit test
  - [ ] Seam / integration test
  - [x] End-to-end test
  - [ ] Existing coverage already sufficient
- Target test or file: `tests/regression/analysis-runner.py`
- Scenario the test should lock in: Steel frame model with custom H-sections completes PKPM analysis without crash, WMASS.OUT shows steel design code GB50017-2017.
- Why this is the smallest reliable guardrail: The analysis regression runner already validates end-to-end PKPM execution.

## User-visible / Behavior Changes

- Steel frame models now produce correct analysis results instead of crashing
- WMASS.OUT material label shows "钢结构" for steel models
- GOUJIAN.OUT has correct section dimensions for custom H-sections

## Diagram (if applicable)

```text
Before:
[V2 JSON with H-sections] -> [wrong field mapping] -> [crash or wrong results]

After:
[V2 JSON with H-sections] -> [correct mapping + Set_M(5)] -> [JWSCYCLE] -> [patch material label] -> [correct WMASS.OUT]
```

## Security Impact (required)

- New permissions/capabilities? No
- Secrets/tokens handling changed? No
- New/changed network calls? No
- Command/tool execution surface changed? No (JWSCYCLE subprocess invocation was already present)
- Data access scope changed? No

### Environment

- OS: Windows 11
- Runtime: Python 3.13 + PKPM2026R2.0 APIPyInterface
- Model/provider: N/A

### Steps

1. Create a 3-story steel frame model with custom H-sections (H400×200×12×20 columns, H350×175×8×14 beams)
2. Run PKPM SATWE analysis via `run_analysis()`
3. Check WMASS.OUT and GOUJIAN.OUT output files

### Expected

- Analysis completes without crash
- WMASS shows "钢结构" and GB50017-2017 design code
- GOUJIAN shows correct section dimensions

### Actual (before fix)

- AddStandFloor crashes with beams
- I-section dimensions stored incorrectly
- Material label shows "钢砼结构"

## Evidence

- [x] Failing test/log before + passing after
- [x] Trace/log snippets

## Human Verification (required)

- Verified scenarios: 3-story steel frame with 8 columns + 10 beams, custom H-sections
- Edge cases checked: Concrete material family (still uses SetUserSect with Set_M(6))
- What I did **not** verify: Multi-standard-floor models, brace elements, non-I-section types

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration

- Backward compatible? Yes
- Config/env changes? No
- Migration needed? No

## Risks and Mitigations

- Risk: Post-processing patch replaces text in WMASS.OUT/WHBJSS.OUT — if JWSCYCLE changes encoding or wording in future versions, the replacement may not work.
  - Mitigation: Patch is conditional on `material_family == "steel"` and only targets the exact GBK bytes for "钢砼结构" → "钢结构".

### CODE REVIEW

File: backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py [hunk 1]
```diff
@@ -132,58 +182,62 @@ def _register_section(
...
-            sec_kind, sh = _make_section_shape(shape_dict)
-            csec.SetUserSect(sec_kind, sh)
+            _sec_kind, sh = _make_section_shape(shape_dict, material_family)
+            csec.SetUserSect(_sec_kind, sh)
         elif std_name:
-            csec.SetStandSteelSect(std_name, APIPyInterface.SectionShape())
+            csec.SetUserSect(
+                APIPyInterface.SectionKind.IDSec_I,
+                APIPyInterface.SectionShape(),
+            )
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py]: Fixed in badf02e — now using `SetStandSteelSect(std_name)` for the `std_name`-only fallback instead of `SetUserSect` with an empty shape.

File: backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py [hunk 2]
```diff
@@ -132,58 +182,62 @@ def _register_section(
...
-            sec_kind, sh = _make_section_shape(shape_dict)
-            bsec.SetUserSect(sec_kind, sh)
+            _sec_kind, sh = _make_section_shape(shape_dict, material_family)
+            bsec.SetUserSect(_sec_kind, sh)
         elif std_name:
-            bsec.SetStandSteelSect(std_name, APIPyInterface.SectionShape())
+            bsec.SetUserSect(
+                APIPyInterface.SectionKind.IDSec_I,
+                APIPyInterface.SectionShape(),
+            )
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py]: Fixed in badf02e — beam section also uses `SetStandSteelSect(std_name)` now.

File: backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py [hunk 3]
```diff
@@ -82,28 +125,35 @@ def _make_section_shape(shape: dict) -> tuple[Any, APIPyInterface.SectionShape]:
...
+    D  = shape.get("D") or shape.get("d")    # V2: diameter (Tube/Circle)
+
+    if sec_kind_attr == "IDSec_I":
+        # PKPM I-section: B=tw, H=height, U=flange_width, T=tf, D=flange_width, F=tf
+        if H  is not None: sh.Set_H(int(H))
+        if tw is not None: sh.Set_B(int(tw))     # web thickness → B
+        if B  is not None: sh.Set_U(int(B))      # flange width  → U (top)
+        if tf is not None: sh.Set_T(int(tf))      # flange thick  → T (top)
+        if B  is not None: sh.Set_D(int(B))      # flange width  → D (bottom, symmetric)
+        if tf is not None: sh.Set_F(int(tf))      # flange thick  → F (bottom, symmetric)
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py]: Fixed in badf02e — changed all `int()` casts to `round()` for section dimensions.

File: backend/src/agent-skills/analysis/pkpm-static/runtime.py [hunk 4]
```diff
@@ -242,27 +327,38 @@ def _extract_results(jws_path: Path) -> Dict[str, Any]:
...
+                    vy = _safe_float(nd.GetDispY())
+                    vz = _safe_float(nd.GetDispZ())
+                    if abs(vx) >= _SENTINEL or abs(vy) >= _SENTINEL or abs(vz) >= _SENTINEL:
+                        continue
+                    pmid = node.GetPmID()
+                    node_key = (pmid, node.GetFloorNo())
+                    case_node_disps.setdefault(case_name, {})[node_key] = {
+                        "ux": abs(vx), "uy": abs(vy), "uz": abs(vz),
+                        "rx": 0.0, "ry": 0.0, "rz": 0.0,
+                    }
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/runtime.py:355]: Fixed in badf02e — removed `abs()` from per-case displacement storage in `case_node_disps` to preserve directionality.

File: backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py [hunk 5]
```diff
@@ -223,6 +277,54 @@ def _build_plan_nodes(
...
+# ---------------------------------------------------------------------------
+
+def _configure_satwe_params(
+    model: APIPyInterface.Model,
+    data: dict,
+    material_family: str,
+    damping_ratio: float = 0.0,
+) -> None:
+    """Set SATWE design parameters via ProjectPara (KIND field codes).
+
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py]: Fixed in badf02e — removed unused `data` and `damping_ratio` parameters from `_configure_satwe_params`.

File: backend/src/agent-skills/analysis/pkpm-static/runtime.py [hunk 6]
```diff
@@ -15,6 +15,7 @@
 from __future__ import annotations
 
 import os
+import re
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/runtime.py]: Fixed in badf02e — removed unused `import re`.

File: backend/src/agent-skills/analysis/pkpm-static/runtime.py [hunk 7]
```diff
@@ -470,7 +733,18 @@ def run_analysis(model: Dict[str, Any], parameters: Dict[str, Any]) -> Dict[str,
...
+    # Build envelope summary with control cases
+    envelope_summary: Dict[str, Any] = {
+        "maxAbsDisplacement": max_disp,
+        "controlCase": {"displacement": ""},
+    }
+    for nid, item in node_displacement_envelope.items():
+        if item["maxAbsDisplacement"] == max_disp:
+            envelope_summary["controlCase"]["displacement"] = item.get("controlCase", "")
+            break
+
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/runtime.py]: Fixed in badf02e — removed dead `envelope_summary` code block.

File: backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py]: Fixed in badf02e — now using `SetStandSteelSect(std_name)` to properly reference the PKPM section library.

File: backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py [hunk 8]
```diff
@@ -373,5 +527,14 @@ def convert_v2_to_jws(
         rf.SetStandFloorIndex(1)
         model.AddNaturalFloor(rf)
 
+    # ---- Configure SATWE design parameters ----
+    _configure_satwe_params(model, data, material_family, damping_ratio)
+    # Diagnostic: log parameter values for index discovery
+    _log_design_params(model)
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/analysis/pkpm-static/pkpm_converter.py]: Fixed in badf02e — gated behind `os.environ.get('PKPM_DEBUG_PARAMS')` so it only runs when explicitly enabled.

File: backend/src/agent-skills/structure-type/frame/extract-llm.ts [hunk 9]
```diff
@@ -72,6 +74,19 @@ function extractAreaLoadIntensity(message: string): number | undefined {
   }, ['direct']);
 }
 
+function extractLiveLoadIntensity(message: string): number | undefined {
+  const livePatterns = [
+    /活载[荷]?\s*[：:]*\s*([0-9]+(?:\.[0-9]+)?)\s*(?:kn|千牛)\s*\/\s*m(?:\^?2|2|²)/i,
+    /live\s*load\s*[：:]*\s*([0-9]+(?:\.[0-9]+)?)\s*(?:kn|千牛)\s*\/\s*m(?:\^?2|2|²)/i,
+    /活荷载\s*[：:]*\s*([0-9]+(?:\.[0-9]+)?)\s*(?:kn|千牛)\s*\/\s*m(?:\^?2|2|²)/i,
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/agent-skills/structure-type/frame/extract-llm.ts:81]: Valid point. Live load tests will be added as a follow-up PR — this change is out of scope for the current PKPM modeling bugfix.

**Review by qinsz01 — APPROVED** [review]
